test/e2e/service.go

```diff
@@ -40,6 +40,10 @@ import (
 	"k8s.io/kubernetes/pkg/util/wait"
 )
 
+// Maximum time a kube-proxy daemon on a node is allowed to not
+// notice a Service update, such as type=NodePort.
+const kubeProxyLagTimeout = 5 * time.Minute
```

Comment:
30s lag is, to me, a pretty significant bug and if it occurs, we should fix
THAT.  I'll argue that service updates should be O(5 seconds) max, but
let's take baby steps.

On Thu, Jan 28, 2016 at 10:45 AM, Prashanth B <notifications@github.com>
wrote:

> In test/e2e/service.go
> <https://github.com/kubernetes/kubernetes/pull/20238#discussion_r51166720>
> :
>
> > @@ -40,6 +40,10 @@ import (
> >  	"k8s.io/kubernetes/pkg/util/wait"
> >  )
> >
> > +// Maximum time a kube-proxy daemon on a node is allowed to not
> > +// notice a Service update, such as type=NodePort.
> > +const kubeProxyLagTimeout = 5 * time.Minute
>
> 30s isn't guaranteed, but yeah. In the specify case I debugged I saw a lag
> of 20s. This is in part kube-proxy inefficiently handling each update of a
> relist of all services one by one, and in part apiserver lag.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20238/files#r51166720>.
>